casa/IO/MultiHDF5.cc

```diff
@@ -37,47 +38,99 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
   MultiHDF5::MultiHDF5 (const String& name, ByteIO::OpenOption option,
                         Int blockSize)
     : MultiFileBase (name, blockSize, False),     //# no O_DIRECT in HDF5
-      itsFile       (itsName, option)
+      itsFile       (new HDF5File(itsName, option)),
+      itsHDF5       (itsFile.get())
+  {
+    init (option);
+  }
+
+  MultiHDF5::MultiHDF5 (const String& name,
+                        const std::shared_ptr<MultiFileBase>& parent,
+                        ByteIO::OpenOption option, Int blockSize)
+    // Use parent's block size if not specified.
+    : MultiFileBase (name, blockSize>0 ? blockSize:parent->blockSize(), False)
+  {
+    // Get the overall HDF5 file object.
+    MultiHDF5* parentHDF5 = dynamic_cast<MultiHDF5*>(parent.get());
+    if (! parentHDF5) {
+      throw AipsError("No MultiHDF5 parent given to nested MultiHDF5 constructor");
+    }
+    itsFile = parentHDF5->getHDF5File();
+    // Create or open the file.
+    // Note that creation (in doAddFile) also creates a dataset which is
+    // not used, but it does not harm.
+    MFFileIO file(parent, name, option, True);
+    itsGroup = file.getInfo().group;    // make sure HDF5Group object is kept
+    itsHDF5  = itsGroup.get();
+    init (option);
+  }
+
+  std::shared_ptr<MultiFileBase> MultiHDF5::makeNested
+  (const std::shared_ptr<MultiFileBase>& parent, const String& name,
+   ByteIO::OpenOption option, Int blockSize) const
+  {
+    return std::shared_ptr<MultiFileBase> (new MultiHDF5 (name, parent,
+                                                          option, blockSize));
+  }
+
+  void MultiHDF5::init (ByteIO::OpenOption option)
   {
     if (option == ByteIO::New  ||  option == ByteIO::NewNoReplace) {
       setNewFile();
     } else {
       readHeader();
     }
-    itsWritable = itsFile.isWritable();
+    itsWritable = itsFile->isWritable();
   }
 
   MultiHDF5::~MultiHDF5()
   {
     close();
   }
 
-  void MultiHDF5::flushFile()
+  void MultiHDF5::doOpenFile (MultiFileInfo& info)
+  {
+    DebugAssert (! info.group->isValid(), AipsError);
+    info.group.reset (new HDF5Group (*itsHDF5, info.name, true, false));
+    info.dataSet.reset (new HDF5DataSet (*info.group, "FileData",
+                                         (const uChar*)0));
+    ///cout << "Opened " << info.name << " in " << itsName << endl;
+  }
+
+  void MultiHDF5::doCloseFile (MultiFileInfo& info)
   {
-    itsFile.flush();
+    DebugAssert (info.group->isValid(), AipsError);
+    info.dataSet.reset();
+    info.group.reset();
+    ///cout << "Closed " << info.name << " in " << itsName << endl;
```

Comment:
Would you like them to be removed?

Reply:
In general I don't like commented out code too much, since its status is unclear.
When there's a good reason to keep it I don't mind.
Is there a log level in Casacore where this output is useful?

Reply:
There is a DEBUG log level, but that's too high.
I've removed them.